src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -326,13 +326,26 @@ private ValueTask<HttpConnection> GetOrReserveHttp11ConnectionAsync(Cancellation
 
             // See if we have an HTTP2 connection
             Http2Connection http2Connection = _http2Connection;
+
+            if (http2Connection != null)
+            {
+                TimeSpan pooledConnectionLifetime = _poolManager.Settings._pooledConnectionLifetime;
```

Comment:
We are testing http2Connection != null below. Can we move this logic there so we don't need to test it twice?